### PR TITLE
Make PS-Committee reviewed experimental features stable

### LINKED ISSUE #15862: Decide on which experimental features will remain in experimental in 7.2
### Summary of the new feature / enhancement

In preparation for 7.2 RCs, @PowerShell/powershell-committee has reviewed our existing experimental features and decided that the following should be upgraded to stable, left as experimental, or removed altogether:

## Supported / Non-Experimental

* `PSNotApplyErrorActionToStderr`: We agree that stderr is used in many cases where there is no error behavior and therefore shouldn't trigger ErrorAction cases like Stop. Given that we've seen no issues with this experimental feature since it was checked into 7.1 previews, we're moving it to stable.
* `PSUnixFileStat`: This has had sufficient time for users in preview builds to use the default behavior of showing Linux file stats (since early 7.1 previews), and no major issues have been reported. 
* `PSAnsiProgress`: This new minimal style of progress bar hasn't created any issues, and we see no reason not to make it stable. Furthermore, you can back to the old progress bar with `$PSStyle.Progress.View = 'classic'`, or turn off progress altogether. 
* `PSAnsiRendering`: There were a few issues around the `$PSStyle` implementation that have been fixed. This points to some amount of usage from customers. Additionally, we've implemented features like `PSAnsiProgress` on top of this, and the interface has been stable and useful. 
* `PSCultureInvariantReplaceOperator`: As outlined in #10948, this issue should make the `-replace` string operator indifferent to the culture in an environment, such that it works the same as the `-split` operator. We haven't seen this cause any impact in the wild from the result of this Bucket 3 breaking change.
* `Microsoft.PowerShell.Utility.PSManageBreakpointsInRunspace`: The `-BreakAll` switch added to `Debug-Runspace` is opt-in, and has been working well within the Azure Functions tooling for Visual Studio Code. 
* `Microsoft.PowerShell.Utility.PSImportPSDataFileSkipLimitCheck`: The `-SkipLimitCheck` switch added to `Import-PowerShellDataFile` is opt-in, and we haven't seen any issues with it in the wild. 

## Experimental (for now)

* `PSAnsiRenderingFileInfo`: This came in very late, and we'd like to give it some more bake time for folks to give feedback. For example, we should continue to discuss the colors themselves, which file extensions should be colorized, how we might be able to support `LS_COLOR`, whether we need theme support, as well as how 3rd parties might be able to extend existing formatters (more context in #15746). To that end, we're leaving it as experimental. 
* `PSCommandNotFoundSuggestion`: The suggestion framework still has significant issues (e.g. writes only to the console host rather than an output stream). This is still something we'd like to leverage in the future for suggesting modules for download from PowerShellGet. 
* `PSNativePSPathResolution`: Today, this resolution of PSPaths in native commands is overly aggressive because single- and double-quotes are not differentiated in the native command binder. This means you can't indicate a PSPath as literal. Until we solve this issue (which is non-trivial), we're leaving this as experimental.
* `PSLoadAssemblyFromNativeCode`: As we discussed in #14641, we are leaving this as experimental until we decide whether or not we want to implement a full native hosting API for wrapping PowerShell from native code.
* `PSSubsystemPluginModel`: Subsystems are still a WIP as we work through the right way to decouple certain functionality from the core PowerShell engine. 

## Needs to be removed

* `PSImplicitRemotingBatching`: This has not fulfilled the original goal of optimizing implicit remoting calls, and has not seen significant usage from folks in the wild. To that end, we plan to revert and remove this code before 7.2 GAs. 

## Deferred

* `PSDesiredStateConfiguration.InvokeDscResource`: Given that this is part of the PSDesiredStateConfiguration module, we are deferring to the module owner to make a final decision on whether this should remain experimental. 

### Proposed technical implementation details (optional)

_No response_

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Each commit makes a feature non-experimental (although one extra due to my filter only looking at .cs files initially...).  Removed the advertising of the experimental feature and any code and tests that checked that the feature is enabled.

To avoid the headache of merge conflicts, I'm submitting this as a single PR.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/15862

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/7917
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/engine/ExperimentalFeature/ExperimentalFeature.cs [hunk 1]
```diff
@@ -122,9 +122,6 @@ static ExperimentalFeature()
                 new ExperimentalFeature(
                     name: "PSNativePSPathResolution",
                     description: "Convert PSPath to filesystem path, if possible, for native commands"),
-                new ExperimentalFeature(
-                    name: "PSNotApplyErrorActionToStderr",
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/ExperimentalFeature/ExperimentalFeature.cs:126]: I found "PSNotApplyErrorActionToStderr" in NativeCommandProcessor.Tests.ps1

Could you confirm that you search and remove all feature names from test files too?

Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/ExperimentalFeature/ExperimentalFeature.cs:126]: I fixed that in another commit already, I don't see any instances in tests

File: src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs [hunk 2]
```diff
@@ -620,13 +620,10 @@ internal void TranscribeResult(Runspace sourceRunspace, string resultText)
 
                     resultText = resultText.TrimEnd();
 
-                    if (ExperimentalFeature.IsEnabled("PSAnsiRendering"))
+                    var text = new ValueStringDecorated(resultText);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:623]: Side note: the struct could be readonly struct.

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs (hunk 2 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:623]: making the struct readonly requires making the fields readonly which breaks the deferred creation of PlainText only when needed

Comment (iSazonov) [reply src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:623]: LazyInitializer could be used for PlainText.

**Comment by SteveL-MSFT** [comment]
CodeQL issue appears to have been fixed recently by GitHub in their action: https://github.com/github/codeql-action/commit/a00769bdfb6706a7400db6aa98570597be644555#diff-5bd808ec74194675da6319beb80c228163039f26c9ddf1b8bf3eef9b12c9341bR490

**Review by rjmholt — APPROVED** [review]

File: src/System.Management.Automation/engine/MshCommandRuntime.cs [hunk 3]
```diff
@@ -2839,7 +2839,7 @@ internal void _WriteErrorSkipAllowCheck(ErrorRecord errorRecord, ActionPreferenc
                 this.PipelineProcessor.LogExecutionError(_thisCommand.MyInvocation, errorRecord);
             }
 
-            if (!(ExperimentalFeature.IsEnabled("PSNotApplyErrorActionToStderr") && isNativeError))
+            if (!isNativeError)
```
Comment (rjmholt) [inline src/System.Management.Automation/engine/MshCommandRuntime.cs:2842]: Ah whoops, this was a mistake... Look at the parens here

**Pull request merged**